move authorize workspace to its home (#2745)
astronomer/docs: astro/authorize-workspaces-to-a-cluster.md

---

> sidebar_label: 'Authorize workspaces to a cluster'
> title: "Authorize Workspaces to a cluster"
> id: authorize-workspaces-to-a-cluster
> description: Authorize your workspace to a cluster

Reviewer: ```suggestion
sidebar_label: 'Authorize Workspaces to a cluster'
title: "Authorize Workspaces to a cluster"
id: authorize-workspaces-to-a-cluster
description: Learn how to configure a cluster so that only specific Workspaces can use it.
```

---

> To provide greater control over your cloud resources, Astro provides you the capability to authorize selective Workspaces to your clusters. This feature is available to Astro Hosted customers using dedicated clusters and Astro Hybrid. For example, you might choose to restrict the dedicated cluster only to your production pipelines within the production Workspace. Meanwhile, your non-production Workspaces can continue to use a standard cluster. 

Reviewer: ```suggestion
To provide greater control over your cloud resources, you can set a rule so that only specific Workspaces can use a dedicated cluster. For example, you can configure a cluster so that only production-level Workspaces can create Deployments in the cluster.
```

> Use this document to understand how to restrict your cluster to selective Workspaces.

Reviewer: ```suggestion
Use this document to learn restrict a cluster so that only authorized Workspaces can use it.
```

## Prerequisites

> - [Organization Owner](user-permissions.md#organization-roles) role
> - [A dedicated cluster on Astro Hosted](create-dedicated-cluster.md) or Astro Hybrid

Reviewer: ```suggestion
- [Organization Owner](user-permissions.md#organization-roles) permissions.
- A [dedicated cluster](create-dedicated-cluster.md).
```

## Authorize your workspace

> 1. In the Cloud UI, click your Workspace name in the upper left corner, then click **Organization Settings**.
> 2. Click **Clusters** in the left pane and then select a cluster.

Reviewer 2: Accessibility nit - folks using screen readers don't perceive left/right/above/below, the UI is organized in before/after, so we just try and avoid using them. Also, the UI design _should_ be sufficient that we only need to call out the UI element text for a user to find what they need when using the docs.
```suggestion
1. In the Cloud UI, click your Workspace name to view more options, then click **Organization Settings**.
2. Click **Clusters** and then select a cluster.
```

3. Go to the **Workspace Authorization** tab and then click **Edit Workspace Authorization**.

> 4. Click **Restricted** and select the Workspaces that you want to give selective access to the cluster.

Reviewer: ```suggestion
4. Click **Restricted** and select the Workspaces that you want to authorize to the cluster.
```

5. Click **Update**.

After you authorize Workspaces to a cluster, Astro treats the cluster as restricted. Restricted clusters do not appear in the dedicated cluster drop-down menu within the **Create Deployment** view of Workspaces that have not been authorized.

> :::tip important

Reviewer: ```suggestion
:::info
```

This is not how we style notes https://docusaurus.io/docs/markdown-features/admonitions

Reviewer: But more importantly, I think this can stay as an `info` note. What's so important about this that a user must know? If they try to restrict a cluster with unauthorized Workspaces, they'll just get an error in the UI telling them what to do. There's nothing a user _has_ to know about before they start this process, because the Cloud UI error handling should cover them.

> Note that a restricted cluster can't host Deployments from any non-authorized Workspaces. If your cluster has Deployments running across multiple Workspaces, and you want to restrict to just one Workspace, you must first transfer the existing Deployments from other Workspaces to the Workspace you want to authorize.

Reviewer: ```suggestion
A restricted cluster can't host Deployments from an unauthorized Workspace. To restrict a cluster that's currently running Deployments from unauthorized Workspaces, you must transfer the Deployments from these Workspaces to the Workspaces you want to authorize.
```

> Similarly, to unauthorize a Workspace but keep its Deployments in the cluster, you must transfer your Deployments to a Workspace which is still authorized to the cluster. See [Transfer a Deployment to another Workspace](configure-deployment-resources.md#transfer-a-deployment-to-another-workspace).

Reviewer 2: ```suggestion
Similarly, to unauthorize a Workspace but still keep its Deployments in the cluster, you must transfer your Deployments to a Workspace that is still authorized to the cluster. See [Transfer a Deployment to another Workspace](configure-deployment-resources.md#transfer-a-deployment-to-another-workspace).
```
